### PR TITLE
PRIME-1880 Email to enrollee after paper submission

### PR DESCRIPTION
**Pull request by AMiKaH**

### CODE REVIEW

**Review by johnnyl6 — APPROVED** [review]

File: prime-dotnet-webapi/ViewModels/Emails/PaperEnrolleeSubmissionEmailViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+using System;
+
+namespace Prime.ViewModels.Emails
+{
+    public class PaperEnrolleeSubmissionEmailViewModel
+    {
+        public string GPID { get; set; }
+
+        public PaperEnrolleeSubmissionEmailViewModel(string gpid) => GPID = gpid;
```
Comment (james-hollinger) [inline prime-dotnet-webapi/ViewModels/Emails/PaperEnrolleeSubmissionEmailViewModel.cs:9]: you do not need a constructor

Comment (AMiKaH) [reply prime-dotnet-webapi/ViewModels/Emails/PaperEnrolleeSubmissionEmailViewModel.cs:9]: This was because I am instantiating this VM in the email email service.

`var email = await _emailRenderingService.RenderPaperEnrolleeSubmissionEmail(enrolleeDto.Email, new PaperEnrolleeSubmissionEmailViewModel(enrolleeDto.GPID));`

Changed to the standard constructor

File: prime-dotnet-webapi/ViewModels/Emails/PaperEnrolleeSubmissionEmailViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+using System;
+
+namespace Prime.ViewModels.Emails
+{
+    public class PaperEnrolleeSubmissionEmailViewModel
+    {
+        public string GPID { get; set; }
+
+        public PaperEnrolleeSubmissionEmailViewModel(string gpid)
```
Comment (hogmann) [inline prime-dotnet-webapi/ViewModels/Emails/PaperEnrolleeSubmissionEmailViewModel.cs:9]: By default when creating a class it has a default constructor that you can access but doing:

```
new PaperEnrolleeSubmissionEmailViewModel
{
    GPID = value;
};
```

So there is not a need to define this constructor and you can use the above to call in your code.

File: prime-dotnet-webapi/Configuration/Database/Seeded Tables/EmailTemplateConfiguration.cs [hunk 3]
```diff
@@ -12,22 +12,23 @@ public override IEnumerable<EmailTemplate> SeedData
...
+                    new EmailTemplate { Id = 17, EmailType = EmailTemplateType.PaperEnrolleeSubmission               , Template = "Your request for PharmaNet access has been approved and recorded in PRIME. When it is possible for you to do so, you must enrol in PRIME using your mobile BC Services Card. <br> <br> <strong> Your temporary GPID is @Model.GPID. </strong> <br> <br> The first time you log into PRIME you should be asked if you have previously received permission to access PharmaNet via an offline process. If you do not see this prompt, please stop your enrollment and contact PRIMESupport@@gov.bc.ca.", ModifiedDate = SEEDING_DATE, CreatedTimeStamp = SEEDING_DATE, UpdatedTimeStamp = SEEDING_DATE}
```
Comment (hogmann) [inline prime-dotnet-webapi/Configuration/Database/Seeded Tables/EmailTemplateConfiguration.cs:31]: Spelling mistake here "PRIMEsupport@@gov.bc.ca" as well as it should be a clickable link.
```<a href='mailto:PRIMEsupport@gov.bc.ca' target='_top'>PRIMEsupport@gov.bc.ca</a>```

**Review by hogmann — CHANGES_REQUESTED** [review]

File: prime-dotnet-webapi/Configuration/Database/Seeded Tables/EmailTemplateConfiguration.cs (hunk 3 above)
Comment (AMiKaH) [reply prime-dotnet-webapi/Configuration/Database/Seeded Tables/EmailTemplateConfiguration.cs:31]: Updated the typo in the file and added and made the link a clickable

File: prime-dotnet-webapi/Migrations/ApiDbContextModelSnapshot.cs [hunk 4]
```diff
@@ -3630,7 +3630,18 @@ protected override void BuildModel(ModelBuilder modelBuilder)
...
+                        },
+                        new
+                        {
+                            Id = 17,
+                            CreatedTimeStamp = new DateTimeOffset(new DateTime(2019, 9, 16, 0, 0, 0, 0, DateTimeKind.Unspecified), new TimeSpan(0, -7, 0, 0, 0)),
+                            CreatedUserId = new Guid("00000000-0000-0000-0000-000000000000"),
+                            EmailType = 17,
+                            ModifiedDate = new DateTimeOffset(new DateTime(2019, 9, 16, 0, 0, 0, 0, DateTimeKind.Unspecified), new TimeSpan(0, -7, 0, 0, 0)),
+                            Template = "Your request for PharmaNet access has been approved and recorded in PRIME. When it is possible for you to do so, you must enrol in PRIME using your mobile BC Services Card. <br> <br> <strong> Your temporary GPID is @Model.GPID. </strong> <br> <br> The first time you log into PRIME you should be asked if you have previously received permission to access PharmaNet via an offline process. If you do not see this prompt, please stop your enrollment and contact <a href=\"mailto:PRIMEsupport@gov.bc.ca",
```
Comment (johnnyl6) [inline prime-dotnet-webapi/Migrations/ApiDbContextModelSnapshot.cs:3644]: The email template seems to be cut off and became incomplete as `<a href=\"mailto:PRIMEsupport@gov.bc.ca"`

Not rendering correctly in mailhog
image

**Review by johnnyl6 — CHANGES_REQUESTED** [review]

File: prime-dotnet-webapi/Migrations/20211112213455_AddedPaperSubmissionTemplate.cs [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+                table: "EmailTemplate",
+                columns: new[] { "Id", "CreatedTimeStamp", "CreatedUserId", "EmailType", "ModifiedDate", "Template", "UpdatedTimeStamp", "UpdatedUserId" },
+                values: new object[] { 17, new DateTimeOffset(new DateTime(2019, 9, 16, 0, 0, 0, 0, DateTimeKind.Unspecified), new TimeSpan(0, -7, 0, 0, 0)), new Guid("00000000-0000-0000-0000-000000000000"), 17, new DateTimeOffset(new DateTime(2019, 9, 16, 0, 0, 0, 0, DateTimeKind.Unspecified), new TimeSpan(0, -7, 0, 0, 0)), "Your request for PharmaNet access has been approved and recorded in PRIME. When it is possible for you to do so, you must enrol in PRIME using your mobile BC Services Card. <br> <br> <strong> Your temporary GPID is @Model.GPID. </strong> <br> <br> The first time you log into PRIME you should be asked if you have previously received permission to access PharmaNet via an offline process. If you do not see this prompt, please stop your enrollment and contact <a href=\"mailto:PRIMEsupport@gov.bc.ca", new DateTimeOffset(new DateTime(2019, 9, 16, 0, 0, 0, 0, DateTimeKind.Unspecified), new TimeSpan(0, -7, 0, 0, 0)), new Guid("00000000-0000-0000-0000-000000000000") });
```
Comment (johnnyl6) [inline prime-dotnet-webapi/Migrations/20211112213455_AddedPaperSubmissionTemplate.cs:38]: The email template seems to be cut off and became incomplete as `<a href=\"mailto:PRIMEsupport@gov.bc.ca"`

Not rendering correctly in mailhog
image

**Review by johnnyl6 — CHANGES_REQUESTED** [review]

File: prime-dotnet-webapi/Migrations/20211112213455_AddedPaperSubmissionTemplate.cs (hunk 5 above)
Comment (AMiKaH) [reply prime-dotnet-webapi/Migrations/20211112213455_AddedPaperSubmissionTemplate.cs:38]: Fixed

Comment (johnnyl6) [reply prime-dotnet-webapi/Migrations/20211112213455_AddedPaperSubmissionTemplate.cs:38]: > Fixed
After running database migration, the email template in DB is still corrupted. see below:

image

Comment (AMiKaH) [reply prime-dotnet-webapi/Migrations/20211112213455_AddedPaperSubmissionTemplate.cs:38]: I did not push my changes yet

Comment (AMiKaH) [reply prime-dotnet-webapi/Migrations/20211112213455_AddedPaperSubmissionTemplate.cs]: No it's updated

**Review by johnnyl6 — APPROVED** [review]

File: prime-dotnet-webapi/Configuration/Database/Seeded Tables/EmailTemplateConfiguration.cs [hunk 6]
```diff
@@ -12,22 +12,23 @@ public override IEnumerable<EmailTemplate> SeedData
...
+                    new EmailTemplate { Id = 17, EmailType = EmailTemplateType.PaperEnrolleeSubmission               , Template = "Your request for PharmaNet access has been approved and recorded in PRIME. When it is possible for you to do so, you must enrol in PRIME using your mobile BC Services Card. <br> <br> <strong> Your temporary GPID is @Model.GPID. </strong> <br> <br> The first time you log into PRIME you should be asked if you have previously received permission to access PharmaNet via an offline process. If you do not see this prompt, please stop your enrollment and contact <a href=\"mailto:PRIMEsupport@gov.bc.ca\" target=\"_top\">PRIMEsupport@gov.bc.ca</a>", ModifiedDate = SEEDING_DATE, CreatedTimeStamp = SEEDING_DATE, UpdatedTimeStamp = SEEDING_DATE}
```
Comment (mtpultz) [inline prime-dotnet-webapi/Configuration/Database/Seeded Tables/EmailTemplateConfiguration.cs:31]: Minor formatting wise pull the semi-colon back, but keeps the spaces

image

Comment (AMiKaH) [reply prime-dotnet-webapi/Configuration/Database/Seeded Tables/EmailTemplateConfiguration.cs]: Fixed

**Review by mtpultz — APPROVED** [review]

File: prime-dotnet-webapi/Migrations/20211113005943_AddedPaperSubmissionTemplate.cs [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+            migrationBuilder.UpdateData(
+                table: "EmailTemplate",
+                keyColumn: "Id",
+                keyValue: 16,
+                column: "Template",
+                value: "Thank you for registering your site (SiteID: @Model.Pec) in PRIME. If you need to update any site information in PRIME, you may log in at any time using your mobile BC Services Card. If you have any questions, please phone 1 - 844 - 397 - 7463 or email <a href=\"mailto:PRIMEsupport@gov.bc.ca\" target=\"_top\">PRIMEsupport@gov.bc.ca</a>. Thank you.");
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Migrations/20211113005943_AddedPaperSubmissionTemplate.cs:33]: Are these updates to existing emails intentional?

File: prime-dotnet-webapi/Migrations/20211113005943_AddedPaperSubmissionTemplate.cs [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+namespace Prime.Migrations
+{
+    public partial class AddedPaperSubmissionTemplate : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.DropIndex(
+                name: "IX_PharmanetTransactionLog_TransactionId",
+                table: "PharmanetTransactionLog");
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Migrations/20211113005943_AddedPaperSubmissionTemplate.cs:12]: You will need to re make this migration, this dropping and re-adding of the index on this table was do to a mistake in another PR.

File: prime-dotnet-webapi/Migrations/20211113005943_AddedPaperSubmissionTemplate.cs (hunk 7 above)
Comment (AMiKaH) [reply prime-dotnet-webapi/Migrations/20211113005943_AddedPaperSubmissionTemplate.cs:33]: Yes, as discussed, I will revert and seek updates from Jason then we can create another ticket for the updates
